contributors/design-proposals/api-machinery/admission-control-webhooks.md

```diff
@@ -254,6 +239,24 @@ consider mechanisms to make this easier. We expect to gather user feedback
 before designing this.
 
 
+## Conversion and Versioning
+
+Webhooks will recieve the admission review subject in the exact version which
+the user sent it to the control plane. This may require the webhook to
+understand multiple versions of those types.
+
+All communication to webhooks will be JSON formatted. We will likely change that
+for GA.
+
+We need to make it possible to know whether an apiserver is safe to upgrade.
+`kube-apiserver` will set the X-Kubernetes-Version header when making any calls
+to a webhook. If the webhook is security-critical, it MUST reject any calls with
+a version greater (as compared by semver) than the most recent one it
+understands. Extension apiservers should also set the X-Kubernetes-Version, to
+be the version of Kubernetes libraries they were built with/from.
+
+This is sort of the minimum we could possibly do here; a design for the next
```

Comment:
I'd drop this and bring it in as an expanding PR

Reply:
I feel like some justification for the above bare-bones system is necessary.